common/src/main/java/whocraft/tardis_refined/common/items/TardisItem.java

```diff
@@ -0,0 +1,76 @@
+package whocraft.tardis_refined.common.items;
+
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.Direction;
+import net.minecraft.core.registries.Registries;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.MutableComponent;
+import net.minecraft.resources.ResourceKey;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.server.level.ServerLevel;
+import net.minecraft.world.InteractionResult;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.Item;
+import net.minecraft.world.item.context.UseOnContext;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.state.BlockState;
+import whocraft.tardis_refined.TardisRefined;
+import whocraft.tardis_refined.command.arguments.DesktopArgumentType;
+import whocraft.tardis_refined.command.arguments.ShellArgumentType;
+import whocraft.tardis_refined.common.tardis.TardisDesktops;
+import whocraft.tardis_refined.common.tardis.themes.DesktopTheme;
+import whocraft.tardis_refined.common.tardis.themes.ShellTheme;
+import whocraft.tardis_refined.common.util.TardisHelper;
+import whocraft.tardis_refined.constants.ModMessages;
+
+import java.util.UUID;
+
+public class TardisItem extends Item {
+    public TardisItem(Properties properties) {
+        super(properties);
+    }
+
+    @Override
+    public InteractionResult useOn(UseOnContext context) {
+
+        Level level = context.getLevel();
```

Comment:
Though the performance impact is negligible, I kinda nitpick with getters that repeat themselves, you retrieve the level up here which is great, but you should then go on to use it in a instanceof check

What I mean by this, is instead of exiting early because we're on the client, instead get the level, then do a ``instanceof ServerLevel level`` approach, then you can store the server as a ``MinecraftServer`` variable